addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,297 @@
+            if move.l10n_tw_edi_is_print or move.l10n_tw_edi_is_donate:
+                move.l10n_tw_edi_carrier_type = False
+                move.l10n_tw_edi_carrier_number = False
+
+    @api.depends('partner_id')
+    def _compute_customer_info(self):
+        for move in self:
+            move.l10n_tw_edi_customer_name = move.partner_id.name
+            move.l10n_tw_edi_customer_address = move.partner_id.contact_address
+            move.l10n_tw_edi_customer_email = move.partner_id.email
+            move.l10n_tw_edi_customer_phone = move.partner_id.phone
+
+    # ----------------
+    # Business methods
+    # ----------------
+
+    # API methods
+
+    def _l10n_tw_edi_prepare_item_list(self):
+        res = []
+        for line in self.invoice_line_ids.filtered(lambda line: line.display_type == 'product' and not line._get_downpayment_lines()):
+            taxable = line.tax_ids.filtered(lambda t: t.amount >= 5.0)
+            item_price = round(line.price_unit, 2)
+            res.append(
+                {
+                    "ItemSeq": line.sequence,
+                    "ItemName": line.product_id.name[:100],
+                    "ItemCount": int(line.quantity),
+                    "ItemWord": line.product_uom_id.name[:6],
+                    "ItemPrice": item_price,
+                    "ItemTaxType": "1" if len(taxable.ids) >= 1 else "3",
+                    "ItemAmount": round(line.price_total, 2),
+                }
+            )
+        return res
+
+    def _l10n_tw_edi_generate_invoice_json(self):
+        if not self.env.company.l10n_tw_edi_ecpay_merchant_id:
+            raise UserError(_("Please fill in the ECpay API information in the Setting!"))
+
+        # prepare json data
+        self.l10n_tw_edi_related_number = base64.urlsafe_b64encode(uuid.uuid4().bytes).decode()
+        json_data = {
+            "MerchantID": self.env.company.l10n_tw_edi_ecpay_merchant_id,
+            "RelateNumber": self.l10n_tw_edi_related_number,
+            "CustomerIdentifier": self.l10n_tw_edi_customer_identifier or "",
+            "CustomerName": self.l10n_tw_edi_customer_name,
+            "CustomerAddr": self.l10n_tw_edi_customer_address,
+            "CustomerPhone": self.l10n_tw_edi_customer_phone or '',
+            "CustomerEmail": self.l10n_tw_edi_customer_email or '',
+            "ClearanceMark": "",
+            "Print": str(int(self.l10n_tw_edi_is_print or self.l10n_tw_edi_customer_identifier)),
+            "Donation": str(int(self.l10n_tw_edi_is_donate)),
+            "TaxType": "1",
+            "SalesAmount": int(self.amount_total),
+            "InvoiceRemark": "Odoo",
+            "InvType": "07",
+            "vat": self.l10n_tw_edi_ecpay_tax_type,
+            "CarrierType": self.l10n_tw_edi_carrier_type or "",
+            "CarrierNum": self.l10n_tw_edi_carrier_number if self.l10n_tw_edi_carrier_type in ["2", "3"] else "",
+            "Items": self._l10n_tw_edi_prepare_item_list()
```

Comment:
To make it easier to extend, it's better to:
1) create json_data without items
2) call _l10n_tw_edi_prepare_item_list and provide the json_data as param of the method, then add the items directly in _l10n_tw_edi_prepare_item_list